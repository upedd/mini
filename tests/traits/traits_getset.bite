native print;

trait A {
    get a;
    set a(value);

    change() {
        a = "foo";
    }

    display() {
       print(a);
    }
}

trait B using A {
    get a {
        print("B.a");
        return "B";
    }

    set a(value) {
        print("=B.a");
    }
}

class C using A {
    a = "C";
}

class D using A {
    _a = "D";
    get a {
        print("D.a");
        return _a;
    }

    set a(value) {
        print("=D.a");
        _a = value;
    }
}

class E using B {
}

let c = C();
c.display();
c.change();
c.display();

let d = D();
d.display();
d.change();
d.display();

let e = E();
e.display();
e.change();
e.display();