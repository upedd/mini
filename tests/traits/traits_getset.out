C
foo
D.a
D
=D.a
D.a
foo
B.a
B
=B.a
B.a
B